store/sagas/fetchMoviesSaga.js

```diff
@@ -0,0 +1,25 @@
+import { put, takeLatest, call, all } from "redux-saga/effects";
+import {
+  FETCH_DEFAULT_MOVIES_FULFILLED,
+  FETCH_DEFAULT_MOVIES_REJECTED,
+  FETCH_DEFAULT_MOVIES_REQUEST,
+} from "../../utils/constant";
+import Api from "../api/api";
+
+function* fetchMovies({ payload }) {
+  try {
+    const { data } = yield call(Api.get, payload);
+    if (data.Response === "True") {
+      yield put({
+        type: FETCH_DEFAULT_MOVIES_FULFILLED,
+        payload: { movies: data.Search, count: data.totalResults },
+      });
+    } else yield put({ type: FETCH_DEFAULT_MOVIES_REJECTED, payload: "error" });
```

Comment:
if your payload is const no need for payload, you can use directly the action type